### PR TITLE
E1 The read should not read the snapshot data into memory

### PR DESCRIPTION
**Pull request by petrabarus**
- [x] Change public attributes to private to hide random access file.
- [x] Refactor Header & Segment for Read and Write.
- [x] Read Header & Segment should read from random access file.

### CODE REVIEW

**Comment by petrabarus** [comment]
@julianghionoiu I'm planning to rename files `SnapshotFile*`, e.g. `SnapshotFileSegment` to `Segment`. Will it break the public API you wrote?

**Comment by julianghionoiu** [comment]
It will not break the public API.

File: src/main/java/tdl/record/sourcecode/snapshot/file/Segment.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
 import tdl.record.sourcecode.snapshot.PatchSnapshot;
 import tdl.record.sourcecode.snapshot.Snapshot;
 
-public class Segment {
+abstract public class Segment {
```
Comment (julianghionoiu) [inline src/main/java/tdl/record/sourcecode/snapshot/file/Segment.java]: I am not sure if inheritance is a good choice here.

I would rather compose the read/write with the base classes.

**Review by julianghionoiu — APPROVED** [review]

**Comment by petrabarus** [comment]
Yes, I think you might be right. Although it does work, I feel a bit uncomfortable with the class design. Let me think of something else.

--
Petra Novandi Barus
https://id.linkedin.com/petrabarus

Sent from Outlook<https://aka.ms/qtex0l> for iOS
________________________________

**Pull request merged**